opm/output/eclipse/summaryComparator.hpp

```diff
@@ -33,59 +34,58 @@ typedef struct ecl_sum_struct ecl_sum_type;
-  stringlist_type* keys1 = nullptr;
-  stringlist_type* keys2 = nullptr; 
-  stringlist_type * keysShort = nullptr;
-  stringlist_type * keysLong = nullptr;
+  ecl_sum_type* ecl_sum1 = nullptr; ///< Struct that contains file1
+  ecl_sum_type* ecl_sum2 = nullptr; ///< Struct that contains file2
+  ecl_sum_type* ecl_sum_fileShort = nullptr; ///< For keeping track of the file with most/fewest timesteps
+  ecl_sum_type * ecl_sum_fileLong = nullptr; ///< For keeping track of the file with most/fewest timesteps
+  stringlist_type* keys1 = nullptr; ///< For storing all the keywords of file1
+  stringlist_type* keys2 = nullptr; ///< For storing all the keywords of file2
+  stringlist_type * keysShort = nullptr; ///< For keeping track of the file with most/fewest keywords
+  stringlist_type * keysLong = nullptr;  ///< For keeping track of the file with most/fewest keywords
   
-  std::vector<double> * referenceVec = nullptr;
-  std::vector<double> * referenceDataVec = nullptr;
-  std::vector<double> * checkVec = nullptr;
-  std::vector<double> * checkDataVec = nullptr;
-
-  void getDeviation(int refIndex, int &checkIndex, Deviation &dev);
-  void setDataSets(std::vector<double> &time_vec1,std::vector<double> &time_vec2);
-  void setTimeVecs(std::vector<double> &timeVec1,std::vector<double> &timeVec2);
-  void getDataVecs(std::vector<double> &dataVec1,std::vector<double> &dataVec2, const char* keyword); 
-void chooseReference(std::vector<double> &timeVec1,std::vector<double> &timeVec2,std::vector<double> &dataVec1,std::vector<double> &dataVec2);
-  double getRelTolerance(){return this->relativeTolerance;}
-  double getAbsTolerance(){return this->absoluteTolerance;}
+  std::vector<double> * referenceVec = nullptr; ///< For storing the values of each time step for the file containing the fewer time steps
+  std::vector<double> * referenceDataVec = nullptr;///< For storing the data values corresponding to each time step for the file containing the fewer time steps
+  std::vector<double> * checkVec = nullptr;///< For storing the values of each time step for the file containing the more time steps
+  std::vector<double> * checkDataVec = nullptr;///< For storing the data values corresponding to each time step for the file containing the more time steps
+  
+  void getDeviation(int refIndex, int &checkIndex, Deviation &dev); /**<Calculates each deviation and stores it in a Deviation struct. Uses the referenceVec as basis, and checks its values against the values int checkDataVec. 
+								     *Should be called inside a for/while loop running over the reference vector, with ivar as the iterative variable of the loop.
+								     *Updates iteration variable jvar aswell, which will be used for the checkVec/cecheDataVec, thus will not check same keyword twice*/
+ 
+  void setDataSets(std::vector<double> &time_vec1,std::vector<double> &time_vec2);///< Figures out which data file that contains the more/fewer time steps, and gives it the correct pointer. 
```

Comment:
better. but one example how i think the doxy should look:
`
//! \brief Figure out which data file contains the most / less timesteps and assign internal pointer accordingly.
//! \param[in] time_vec1 Data from first file
//! \param[in] time_vec2 Data from second file
`